dockercheck

```diff
@@ -1,66 +1,113 @@
+DISK_USAGE_THRESHOLD = int(os.environ.get("DOCKERCHECK_DISK_USAGE_THRESHOLD", "90"))
 
-prom_container_status = Enum(METRIC_NAME, 'Returns the status of a Compose project container',
-        states=['restarting', 'running', 'paused', 'exited', 'unhealthy'], labelnames=["name", "role"])
+prom_container_status = Enum(
+    METRIC_NAME,
+    "Returns the status of a Compose project container",
+    states=["restarting", "running", "paused", "exited", "unhealthy"],
+    labelnames=["name", "role"],
+)
 
-def check_containers_status(client, filter_label):
-    '''
-    Check the containers statuses and export metrics
+
+def check_disk_usage(threshold):
+    """
+    Checks the disk usage and returns True if it's below the specified threshold,
+    otherwise returns False.
+    """
+    try:
+        disk_usage_percent = psutil.disk_usage("/").percent
+        if disk_usage_percent > threshold:
+            logging.error(f"Disk usage is over {threshold}%: {disk_usage_percent}%")
+            return False
+    except Exception as e:
+        logging.error(f"Error checking disk usage: {e}")
+        return False
+    return True
+
+
+def check_containers_status(client, filter_label, disk_usage_threshold):
+    """
+    Check the containers statuses and disk usage and export metrics
 
     Iterates over a list of containers in *any* state and filtering on `filter_label`
-    '''
-    containers = client.containers.list(all=True,
-                                        filters={"label": filter_label})
+    Updates Prometheus metrics for each container and logs disk usage errors.
+
+    It returns HTTP status code (200) if all containers are running and disk usage is below threshold.
+
+    """
+    containers = client.containers.list(all=True, filters={"label": filter_label})
     if not containers:
-        return(500)
+        return 500
 
-    http_status=200
+    http_status = 200
     for container in containers:
-        prom_container_status.labels(container.name,container.labels[filter_label]).state(container.status)
+        prom_container_status.labels(
+            container.name, container.labels[filter_label]
+        ).state(container.status)
         if container.status != "running":
-            http_status=500
+            http_status = 500
         try:
             if container.attrs["State"]["Health"]["Status"] == "unhealthy":
                 # health can be "starting", "healthy" and "unhealthy". we care just about the latter
-                prom_container_status.labels(container.name,container.labels[filter_label]).state("unhealthy")
-                http_status=500
+                prom_container_status.labels(
+                    container.name, container.labels[filter_label]
+                ).state("unhealthy")
+                http_status = 500
         except KeyError:
             pass
-    return(http_status)
+    # Check disk usage
+    if not check_disk_usage(disk_usage_threshold):
```

Comment:
I like having the function returning a bool but can you please ivnert the condition, maybe even renaming the function? Like name it `check_disk_full` and then it returns `True` if full, and you can get rid of the `not` here